lib/Basics/ReadWriteLock.cpp

```diff
@@ -35,11 +35,10 @@ void ReadWriteLock::lockWrite() {
 
   // the lock is either hold by another writer or we have active readers
   // -> announce that we want to write
-  _state.fetch_add(QUEUED_WRITER_INC, std::memory_order_relaxed);
+  auto state = _state.fetch_add(QUEUED_WRITER_INC, std::memory_order_relaxed);
 
   std::unique_lock<std::mutex> guard(_writer_mutex);
   while (true) {
-    auto state = _state.load(std::memory_order_relaxed);
```

Comment:
It was intentional to reload `_state` after acquiring the mutex, because in case we were blocked (because someone else was holding the mutex), `_state` most likely has changed, causing the subsequent CAS to fail. If we were not blocked, then the additional load will most certainly hit the L1 cache and should therefore be very cheap.

Reply:
Ok, let me change that back and add this comment to the code.

Reply:
Fixed